### DISCUSSION TITLE
Can you guys help me?

### DISCUSSION THREAD

**Discussion by phamt8611** [termux/termux-app#2439] [Q&A] [answered]
Can you guys help me?
When I write "pip install pyaudio" the error message

`  ~ $ pip install pyaudio
Collecting pyaudio
  Downloading PyAudio-0.2.11.tar.gz (37 kB)
  Preparing metadata (setup.py) ... done
Using legacy 'setup.py install' for pyaudio, since package 'wheel' is not installed.
Installing collected packages: pyaudio
    Running setup.py install for pyaudio ... error
    ERROR: Command errored out with exit status 1:
     command: /data/data/com.termux/files/usr/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-b09yetl0/pyaudio_de5e3a5ac2444887a54d618f37c0cd71/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-b09yetl0/pyaudio_de5e3a5ac2444887a54d618f37c0cd71/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /data/data/com.termux/files/usr/tmp/pip-record-s1gd74qj/install-record.txt --single-version-externally-managed --compile --install-headers /data/data/com.termux/files/usr/include/python3.10/pyaudio
         cwd: /data/data/com.termux/files/usr/tmp/pip-install-b09yetl0/pyaudio_de5e3a5ac2444887a54d618f37c0cd71/
    Complete output (18 lines):
    running install
    /data/data/com.termux/files/usr/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.10
    copying src/pyaudio.py -> build/lib.linux-aarch64-3.10
    running build_ext
    building '_portaudio' extension
    creating build/temp.linux-aarch64-3.10
    creating build/temp.linux-aarch64-3.10/src
    aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.10 -c src/_portaudiomodule.c -o build/temp.linux-aarch64-3.10/src/_portaudiomodule.o
    src/_portaudiomodule.c:29:10: fatal error: 'portaudio.h' file not found
    #include "portaudio.h"
             ^~~~~~~~~~~~~
    1 error generated.
    error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /data/data/com.termux/files/usr/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-b09yetl0/pyaudio_de5e3a5ac2444887a54d618f37c0cd71/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-b09yetl0/pyaudio_de5e3a5ac2444887a54d618f37c0cd71/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /data/data/com.termux/files/usr/tmp/pip-record-s1gd74qj/install-record.txt --single-version-externally-managed --compile --install-headers /data/data/com.termux/files/usr/include/python3.10/pyaudio Check the logs for full command output.`

**Comment by 19akshansh** [comment, marked as answer]
**Hello there!**

You need to setup "its-pointless" repo to install pyaudio on your termux.

To setup "its-pointless" repo you can either use "curl" or use "wget"


Here are both of the commands, select the one you want/often use.

# WGET :-
```
wget https://raw.githubusercontent.com/its-pointless/its-pointless.github.io/master/setup-pointless-repo.sh && bash setup-its-pointless-repo.sh
```

# CURL :-
```
curl -sL https://raw.githubusercontent.com/its-pointless/its-pointless.github.io/master/setup-pointless-repo.sh | bash
```


Btw, I prefer & use curl!


# IMPORTANT EDITED :-

once "its-pointless" repo's setup is done, use

```
apt install portaudio
```

& then install "pyaudio" with 

```
pip install pyaudio
```

**Comment by phamt8611** [comment]
thank you

**Reply by 19akshansh** [reply]
Your Welcome.
